imports/server/hooks/HooksRegistry.ts

```diff
@@ -21,39 +21,35 @@ class HooksRegistry {
   }
 
   async runPuzzleCreatedHooks(puzzleId: string) {
-    await this.registeredHooks.reduce(async (p, hook) => {
-      await p;
+    for (const hook of this.registeredHooks) {
```

Comment:
This is orthogonal to this particular patchset, but it occurs to me that this file is probably one place we should take the hint from the lint and fire off hooks in parallel and do an `await Promise.all` for each of these here.